packages/neon_framework/test/request_manager_test.dart

```diff
@@ -0,0 +1,927 @@
+// ignore_for_file: unnecessary_lambdas
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:neon_framework/src/bloc/result.dart';
+import 'package:neon_framework/src/utils/request_manager.dart';
+import 'package:nextcloud/nextcloud.dart';
+import 'package:rxdart/rxdart.dart';
+
+class MockCallbackFunction<T> extends Mock {
+  FutureOr<T> call();
+}
+
+class MockedCache extends Mock implements Cache {}
+
+void main() {
+  tearDown(() {
+    RequestManager.instance = null;
+    Cache.instance = null;
+  });
+
+  group('Cache', () {
+    test('singleton', () {
+      expect(identical(Cache(), Cache()), isTrue);
+    });
+  });
+
+  group('RequestManager', () {
+    test('singleton', () {
+      expect(identical(RequestManager.instance, RequestManager.instance), isTrue);
+    });
+
+    test('timeout', () async {
+      final callback = MockCallbackFunction<bool>();
+      when(callback.call).thenAnswer((_) async => Future.delayed(const Duration(milliseconds: 100), () => true));
+
+      expect(
+        () => RequestManager.instance.timeout<bool>(
+          () async => callback.call(),
+          timeLimit: const Duration(milliseconds: 50),
+        ),
+        throwsA(isA<TimeoutException>()),
+      );
+      verify(callback.call).called(1);
+
+      var result = await RequestManager.instance.timeout<bool>(
+        () async => callback.call(),
+        disableTimeout: true,
+        timeLimit: const Duration(milliseconds: 50),
+      );
+      expect(result, isTrue);
+      verify(callback.call).called(1);
+
+      result = await RequestManager.instance.timeout<bool>(
+        () async => callback.call(),
+        timeLimit: const Duration(milliseconds: 150),
+      );
+      expect(result, isTrue);
+      verify(callback.call).called(1);
+    });
+
+    group('wrap without cache', () {
+      test('successful request', () async {
+        var subject = BehaviorSubject<Result<String>>();
+
+        // ignore: unawaited_futures
+        expectLater(
+          subject.stream,
+          emitsInOrder([
+            equals(Result<String>.loading()),
+            equals(Result.success('VGVzdCB2YWx1ZQ==')),
```

Comment:
Please do the encoding on the fly so when reading the test you can know what the value is